assets/ui/components/ArticleBodyHtml.tsx

```diff
@@ -187,49 +215,27 @@ class ArticleBodyHtml extends React.PureComponent<any, any> {
             });
 
         // If Image tags were not updated, then return the supplied html as-is
-        return imageSourcesUpdated ?
-            container.innerHTML :
-            html;
+        return imageSourcesUpdated ? container.innerHTML : html;
     }
 
     render() {
-        const item = this.props.item;
-        const html = this.getBodyHTML(
-            get(item, 'es_highlight.body_html.length', 0) > 0 ?
-                item.es_highlight.body_html[0] :
-                item.body_html
-        );
-
-        if (!html) {
-            return null;
-        }
+        // preview element will be populated in `componentDidMount`
 
         return (
             <div
                 ref={this.bodyRef}
                 className='wire-column__preview__text wire-column__preview__text--pre'
                 id='preview-body'
-                dangerouslySetInnerHTML={({__html: html})}
             />
         );
     }
 }
 
-ArticleBodyHtml.propTypes = {
-    item: PropTypes.shape({
-        body_html: PropTypes.string,
-        es_highlight: PropTypes.shape({
-            body_html: PropTypes.arrayOf(PropTypes.string),
-        }),
-        associations: PropTypes.object,
-    }).isRequired,
-    reportCopy: PropTypes.func,
-};
-
 const mapDispatchToProps = (dispatch: any) => ({
     reportCopy: (item: any) => dispatch(selectCopy(item))
 });
 
-const component: React.ComponentType<any> = connect(null, mapDispatchToProps)(ArticleBodyHtml);
+const ArticleBodyHtmlConnected: React.ComponentType<IOwnProps> = connect(null, mapDispatchToProps)(ArticleBodyHtmlComponent);
 
-export default component;
+// component needs to reinitialize if item changes
+export const ArticleBodyHtml: React.ComponentType<IOwnProps> = (props) => <ArticleBodyHtmlConnected {...props} key={props.item._id} />;
```

Comment:
Using `key=item._id` should be left up to the container component that renders this one, otherwise we're adding an extra branch in the render tree just to add a `key` property.

Reply:
removed that

Reply:
Normally, components are expected to react to prop changes and update accordingly. Sometimes code-wise it is simpler to ignore prop changes and to force-reinitialize the component using a key. This is an implementation detail, thus a parent component shouldn't have to know about it.

I don't mind that you changed it Petr, I'm posting this comments so everyone understands the reasoning and we have an opportunity to discuss the pattern.